execution-engine/contract-ffi/src/value/protocolversion.rs

```diff
@@ -0,0 +1,60 @@
+use uint::core_::fmt;
+
+#[derive(Debug, Default, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub struct ProtocolVersion(u64);
+
+impl ProtocolVersion {
+    pub fn new(version: u64) -> ProtocolVersion {
+        ProtocolVersion(version)
+    }
+
+    pub fn get(self) -> u64 {
+        self.0
+    }
+}
+
+impl fmt::Display for ProtocolVersion {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self.0)
```

Comment:
Does it make sense to write a `Display` impl if just delegates to the `Debug` printer?  

cc @Fraser999 

If so, it seems like you could just:

```suggestion
        Debug::fmt(self, f, fmt)
```

Reply:
We do this in various places currently...

![image](https://user-images.githubusercontent.com/49160758/65084552-679de680-d960-11e9-9a75-6c15142eda2e.png)

Reply:
I'm not sure if there's too much point to having a `Display` and `Debug` implemented if they both produce the same output.  I guess a minor point is that if `Display` is implemented on a type, [we get `to_string()` for free](https://doc.rust-lang.org/std/string/trait.ToString.html) too.

However, the original code isn't quite just delegating to the `Debug` printer, it's `self.0` which is being used, so with the original code we'd have e.g. `ProtocolVersion(670)` and `670` for `Debug` and `Display` respectively.  Seems like having an option to lose the `ProtocolVersion` from the output could be worthwhile if we're printing a struct where the member is well-named and the extra characters would just be noise.

Reply:
> However, the original code isn't quite just delegating to the Debug printer, it's self.0 which is being used.

Whoops, good point, I missed that.

Reply:
Alright, seems reasonable for now.  We might audit the cases where we are not using `self.0`.